plugin/test-keeper/plugin/event_handler_test.go

```diff
@@ -0,0 +1,180 @@
+
+			// then - implicit verification of /statuses call occurrence with proper payload
+		})
+
+		It("should block newly created pull request when no tests are included", func() {
+
+			// given
+			gock.New("https://api.github.com").
+				Get("/repos/bartoszmajsak/wfswarm-booster-pipeline-test/pulls/1/files").
+				Reply(200).
+				Body(fromJson("test_fixtures/github_calls/prs/without_tests/changes.json"))
+
+
+			toHaveFailureState := func(statusPayload map[string]interface{}) (bool) {
+				return Expect(statusPayload["state"]).To(Equal("failure"))
+			}
+
+			gock.New("https://api.github.com").
+				Post("/repos/bartoszmajsak/wfswarm-booster-pipeline-test/statuses").
+				SetMatcher(expectStatusCall(toHaveFailureState)).
+				Reply(201) // This way we implicitly verify that call happened after `HandleEvent` call
+
+			// when
+			handler.HandleEvent(githubevents.PullRequest, eventGUID, eventPayload("test_fixtures/github_calls/prs/without_tests/status_opened.json"))
+
+			// then - implicit verification of /statuses call occurrence with proper payload
+		})
+
+		It("should skip test existence check when "+plugin.SkipComment+" command is used by admin user", func() {
+
+			// given
+			gock.New("https://api.github.com").
+				Get("/repos/bartoszmajsak/wfswarm-booster-pipeline-test/pulls/1").
+				Reply(200).
+				Body(fromJson("test_fixtures/github_calls/prs/without_tests/pr_details.json"))
+
+			gock.New("https://api.github.com").
+				Get("/repos/bartoszmajsak/wfswarm-booster-pipeline-test/collaborators/bartoszmajsak/permission").
+				Reply(200).
+				Body(fromJson("test_fixtures/github_calls/collaborators_bartoszmajsak_permission.json"))
+
+			toHaveSuccessState := func(statusPayload map[string]interface{}) (bool) {
+				return Expect(statusPayload["state"]).To(Equal("success"))
+			}
+
+			gock.New("https://api.github.com").
+				Post("/repos/bartoszmajsak/wfswarm-booster-pipeline-test/statuses").
+				SetMatcher(expectStatusCall(toHaveSuccessState)).
+				Reply(201) // This way we implicitly verify that call happened after `HandleEvent` call
+
+			// when
+			handler.HandleEvent(githubevents.IssueComment, eventGUID, eventPayload("test_fixtures/github_calls/prs/without_tests/skip_comment_by_admin.json"))
+
+			// then - implicit verification of /statuses call occurrence with proper payload
+		})
+
+		It("should ignore "+plugin.SkipComment+" when used by non-admin user", func() {
+			// given
+			gock.New("https://api.github.com").
+				Get("/repos/bartoszmajsak/wfswarm-booster-pipeline-test/pulls/1").
+				Reply(200).
+				Body(fromJson("test_fixtures/github_calls/prs/without_tests/pr_details.json"))
+
+			gock.New("https://api.github.com").
+				Get("/repos/bartoszmajsak/wfswarm-booster-pipeline-test/collaborators/bartoszmajsak-test/permission").
+				Reply(200).
+				Body(fromJson("test_fixtures/github_calls/collaborators_bartoszmajsak-test_permission.json"))
```

Comment:
the files
`collaborators_bartoszmajsak-test_permission.json`
`collaborators_bartoszmajsak_permission.json`
have very similar names (which is very confusing) and it is not obvious what they represent. 
btw, firstly I read the first file as `collaborators_bartoszmajsak_test_permission.json` which confused me totally.
It would be good to rename the files (or the users) to have something more meaningful:
`bartoszmajsak-admin`
`bartoszmajsak-non-admin`
or maybe better:
`ike-admin`
`ike-non-admin` 
without the specific usernames.
The same applies to the rest of the files and calls. I know why you put it there - you just copy-pasted the responses, but it is quite hard to modify/reproduce it for the other users. Mainly because it is releated to your personal account, repository and PRs, that can by modified/removed by you anytime.

Reply:
As we discussed last week, long term goal would be to look at `go-vcr` or `hoverfly`, even though I like `gock` dsl for stubbing http calls. But even with that we will have to make recordings against real GH use case (users and repo).

I will rename the files to make it a bit less cryptic and let's think about overall improvement as we progress with the implementations.

Reply:
> The same applies to the rest of the files and calls. I know why you put it there - you just copy-pasted the responses, but it is quite hard to modify/reproduce it for the other users. Mainly because it is releated to your personal account, repository and PRs, that can by modified/removed by you anytime.

I'm open to better names, but keep in mind it's a stub at this point and it is unlikely to be changed any time.

I renamed permission `json`s in 9d80b46. I hope it's slightly better now.